pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -345,6 +348,68 @@ public SuccessResponse updateTableConfig(
 
     return new SuccessResponse("Table config updated for " + tableName);
   }
+  
+  /**
+   * Truncate table, delete all contents of table without removing table configuration and schema. 
+   * 
+   * @param tableName , name of the table to truncate
+   * @param tableTypeStr, type of the table i.e 'offline|realtime'
+   * @return response of truncate operation.
+   * @throws Exception
+   */
+  @POST
+  @Path("/tables/{tableName}/truncate")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Truncate table")
+  public SuccessResponse truncateTable(
+      @ApiParam(value = "Name of the table to update", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr) throws Exception {
+
+    TableType tableType = Constants.validateTableType(tableTypeStr);
+    if (tableType == null) {
+      throw new ControllerApplicationException(LOGGER, "Table type must not be null", Response.Status.BAD_REQUEST);
+    }
+
+    // Disable table by tableType
+    toggleTableState(tableName, tableType, StateType.DISABLE);
+
+    // Get all segment names for table and delete all segments
+    String tableNameWithType = getExistingTableNamesWithType(tableName, tableType).get(0);
```

Comment:
How about moving line `String tableNameWithType = getExistingTableNamesWithType(tableName, tableType).get(0);` before the toggle, so that we are validating if table exists before calling disable.

Then you can also directly call `_pinotHelixResourceManager.toggleTableState(tableNameWithType, status);`